From e293716c211468da779e65b8860430c055de78fa Mon Sep 17 00:00:00 2001
From: Topi Reinio <topi.reinio@qt.io>
Date: Wed, 18 Nov 2020 11:37:07 +0100
Subject: [PATCH 13/17] qdoc: Avoid excess warnings for undocumented namespaces

There is no need to generate a warning for an undocumented namespace
that is documented in another module.

Fixes: QTBUG-88603
Change-Id: Ibcaecad32a165776cd32290c823948570e3556cf
Reviewed-by: Paul Wicking <paul.wicking@qt.io>
(cherry picked from commit 9eb640775743efe477d7791f7440a313447b9f61)
---
 src/qdoc/htmlgenerator.cpp | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/src/qdoc/htmlgenerator.cpp b/src/qdoc/htmlgenerator.cpp
index a554f1ac..24735237 100644
--- a/src/qdoc/htmlgenerator.cpp
+++ b/src/qdoc/htmlgenerator.cpp
@@ -1286,8 +1286,12 @@ void HtmlGenerator::generateCppReferencePage(Aggregate *aggregate, CodeMarker *m
         QString command = "documentation";
         if (aggregate->isClassNode())
             command = "\'\\class\' comment";
-        aggregate->location().warning(
-                tr("No %1 for '%2'").arg(command).arg(aggregate->plainSignature()));
+        if (!ns || ns->isDocumentedHere()) {
+            aggregate->location().warning(
+                    tr("No %1 for '%2'")
+                        .arg(command)
+                        .arg(aggregate->plainSignature()));
+        }
     } else {
         generateExtractionMark(aggregate, DetailedDescriptionMark);
         out() << "<div class=\"descr\">\n" // QTBUG-9504
-- 
2.31.1

